src/core/utils/filternodeutils.js

```diff
@@ -0,0 +1,41 @@
+import FilterNodeFactory from '../filters/filternodefactory';
+import Filter from '../models/filter';
+import FilterMetadata from '../filters/filtermetadata';
+
+/**
+ * Converts an array of {@link AppliedQueryFilter}s into equivalent {@link SimpleFilterNode}s.
+ * @param {Array<AppliedQueryFilter>} nlpFilters
+ * @returns {Array<SimpleFilterNode>}
+ */
+export function convertNlpFiltersToFilterNodes (nlpFilters) {
+  return nlpFilters.map(nlpFilter => FilterNodeFactory.from({
+    filter: Filter.from(nlpFilter.filter),
+    metadata: new FilterMetadata({
+      fieldName: nlpFilter.key,
+      displayValue: nlpFilter.value
+    })
+  }));
+}
+
+export function flattenIntoSimpleFilterNodes (filterNodes) {
```

Comment:
I think just calling this "flatten" and providing JSDoc would be enough. This name is kind of long.

Reply:
how about `flattenFilterNodes()` ? I feel like just calling it `flatten()` makes it hard to search for and could imply it's flattening a generic array